dask_xgboost/core.py

```diff
@@ -202,14 +228,26 @@ def _train(client, params, data, labels, dmatrix_kwargs={}, **kwargs):
 
     # Get the results, only one will be non-None
     results = yield client._gather(futures)
-    result = [v for v in results if v][0]
+    result, _evals_result = [v for v in results if v.count(None) != len(v)][0]
+
+    if evals_result is not None:
+        evals_result.update(_evals_result)
```

Comment:
Just to confirm: we mutate the user-provided `evals_result` dictionary inplace, which matches the behavior of xgboost?

Reply:
correct